aten/src/ATen/NestedTensorImpl.h

```diff
@@ -56,6 +72,51 @@ struct TORCH_API NestedTensorImpl : public c10::TensorImpl {
   const at::Tensor nested_size_tensor_;
   // NOTE: -1 here means the size is missing
   std::vector<int64_t> opt_sizes_;
+
+  // Implicit batch dimension index offset
```

Comment:
@bhosmer yes I talked to Joel after our conversation, we agreed it would be nice to have the batch dimension offset as a separate utility, since it will be called again and again

Reply:
@YifanShenSZ nice! 

Also note that to build a `nested_stride_tensor` you'd just be doing [the same product loop](https://github.com/pytorch/pytorch/pull/79831/files#diff-ce5725da7d5407b5a3c6342bc2b2a640daf3d78c4291c7c34327e57db0d33ad9R89) but writing out the intermediate multiples. So if/when you do strides, it'd probably pay to set up both structures at once...

BTW the subject of strides came up at the sparsity meeting yesterday - sounds like it won't be an issue for autograd as long as `supports_as_strided()` returns false for NTs. (See the comment [here](https://github.com/pytorch/pytorch/blob/master/c10/core/TensorImpl.h#L1029-L1041), also Alban will have more details, he tipped us off to the distinction 😁 )

Reply:
Christian also suggests to add `nested_stride_tensor` in this pr, so yeah [here](https://github.com/pytorch/pytorch/pull/79831/files#diff-e9993081feee11d32a3228d428f8bc33f15bd3956d50d9bfd48bdac807e628f7) it is.

I saw Driss's autograd pr last week then consulted Alban and Driss (thx Alban and Driss!). Letting `supports_as_strided` return false means to let autograd engine use a slower but general way: record all view functions. So we won't have any trouble in defining `nested_stride_tensor`